analysis_driver/client.py

```diff
@@ -98,6 +105,24 @@ def _process_dataset(d):
         (AsanaNotification, d, cfg.query('notification', 'asana'))
     )
 
+    def _handle_exception(exception):
+        import traceback
+        stacktrace = traceback.format_exc()
+        app_logger.critical('Encountered a %s exception: %s', exception.__class__.__name__, str(exception))
+        app_logger.info('Stack trace below:\n' + stacktrace)
+        _handle_termination(9)
+        ntf.crash_report(stacktrace)
```

Comment:
It looks like there's only a stacktrace to log if there's a `KeyboardInterrupt` (signal 2).

`_handle_termination` will call notifications via `dataset.fail` with the signal number as the exit status. Is this enough information, or would we like an explicit `dataset.sigterm` (or whatever)?